bin/validate_exercises

```diff
@@ -8,7 +8,15 @@ set -o nounset
 
 cd "$( dirname "${BASH_SOURCE[0]}" )/.."
 
-for exercise in exercises/*/*; do
+while read -r exercise; do
```

Comment:
Is the config better than checking what files are there?
This messes with the STDIN of the validate script.

```suggestion
mapfile -t exercises < <(
    jq -r '
        .exercises  as $exs 
        | ["concept","practice"][] 
        | . as $type 
        | $exs[$type][] 
        | "exercises/\($type)/\(.slug)"
    ' config.json
)

for exercise in "${exercises[@]}"; do
    bin/validate_one_exercise "$exercise"
    echo
done
```

Reply:
In my awk repo, I still had the (now foregone) circular-buffer exercise, and that broke the testing. This was how I worked around it.

But what I really want is to compare what's in exericses/ and what's in config.json, and warn about differences.

I'll take your suggestion and add to it.